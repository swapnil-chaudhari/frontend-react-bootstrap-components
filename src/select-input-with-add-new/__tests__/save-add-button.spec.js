import React from 'react';
import SaveAddButton from '../save-add-button';
import renderShallow from 'render-shallow';

describe('<SaveAddButton>', () => {

    describe('when it renders with required props', () => {
        const props = {
            text: 'Add me',
            name: 'artist',
            onClick: jest.fn()
        };
        let component;

        beforeAll(() => {
            component = renderShallow(<SaveAddButton { ...props } />).output;
        });

        test('renders a button with an onClick and the right text', () => {
            expect(component).toMatchSnapshot();
        });

    });

    describe('when props.disabled is passed in and true', () => {
        const props = {
            text: 'Add artist',
            disabled: true,
            onClick: jest.fn()

        };
        let component;

        beforeAll(() => {
            component = renderShallow(<SaveAddButton { ...props } />).output;
        });

        test('sets the button.disabled to true', () => {
            expect(component.props.disabled).toBeTruthy();
        });

    });

    describe('when props.disabled is passed in and false', () => {
        const props = {
            disabled: false,
            text: 'Add artist',
            onClick: jest.fn()

        };
        let component;

        beforeAll(() => {
            component = renderShallow(<SaveAddButton { ...props } />).output;
        });

        test('sets the button.disabled to false', () => {
            expect(component.props.disabled).toBeFalsy();
        });

    });

});
