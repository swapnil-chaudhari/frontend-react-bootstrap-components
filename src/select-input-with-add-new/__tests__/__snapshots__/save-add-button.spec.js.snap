// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SaveAddButton> when it renders with required props renders a button with an onClick and the right text 1`] = `
<Button
  disabled={undefined}
  onClick={[MockFunction]}
>
  Add me
</Button>
`;
